featurebyte/worker/task/base.py

```diff
@@ -25,6 +27,61 @@ class BaseTask(Generic[TaskT]):
 
     payload_class: Type[TaskT]
 
+    def __init__(self, task_manager: TaskManager):
+        self.task_manager = task_manager
+
+        # track the task ID
+        self._task_id: Optional[TaskId] = None
+
+    @property
+    def task_id(self) -> Optional[TaskId]:
+        """
+        Task ID
+
+        Returns
+        -------
+        Optional[TaskId]
+        """
+        return self._task_id
+
+    def set_task_id(self, task_id: TaskId) -> None:
+        """
+        Set task ID
+
+        Parameters
+        ----------
+        task_id: TaskId
+            Task ID
+        """
+        self._task_id = task_id
+
+    async def submit_task(
```

Comment:
Maybe can name this as `submit_child_task` to make its behaviour clearer vs submitting via task manager's submit.